src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -321,15 +458,20 @@ void ZeroInferRequest::infer_async() {
             }
         }
 
-        const uint8_t* tensorBuffer = reinterpret_cast<uint8_t*>(inputTensor->data());
-        uint8_t* copyTensorBuffer = reinterpret_cast<uint8_t*>(wrapperInputTensor->data());
+        auto remoteTensor = std::dynamic_pointer_cast<ZeroRemoteTensor>(inputTensor);
+        if (remoteTensor == nullptr) {
+            const auto& wrapperInputTensor = _copyAllTensors.at(name);
 
-        if (tensorBuffer != copyTensorBuffer) {
-            if (tensorBuffer == nullptr || copyTensorBuffer == nullptr) {
-                OPENVINO_THROW("Empty buffer");
-            }
+            const uint8_t* tensorBuffer = reinterpret_cast<uint8_t*>(inputTensor->data());
+            uint8_t* copyTensorBuffer = reinterpret_cast<uint8_t*>(wrapperInputTensor->data());
+
+            if (tensorBuffer != copyTensorBuffer) {
+                if (tensorBuffer == nullptr || copyTensorBuffer == nullptr) {
+                    OPENVINO_THROW("Empty buffer");
+                }
 
-            std::memcpy(copyTensorBuffer, tensorBuffer, inputTensor->get_byte_size());
+                std::memcpy(copyTensorBuffer, tensorBuffer, inputTensor->get_byte_size());
```

Comment:
The Remote Tensor design was meant to eliminate this implicit memcpy, see [here](https://intel-my.sharepoint.com/:p:/p/javier_e_martinez/EYHDkY0t37VFiWLRE9GUhgUBah54bzKkbLh3Q7lyylw-Fw?e=N6mP9Z&nav=eyJzSWQiOjQxMywiY0lkIjoxMDQ3Mjg1NzA2fQ).

Reply:
In case setTensor is used with a normal tensor, we still need this memcpy

Reply:
You can also extend the logger message to say that an extra copy is introduced.

Reply:
Yeah, just added a loogger there :).